java/spark/v3/src/main/java/com/lancedb/lance/spark/query/FilterPushDown.java

```diff
@@ -0,0 +1,167 @@
+    if (filter instanceof GreaterThan) {
+      GreaterThan f = (GreaterThan) filter;
+      return Optional.of(f.attribute() + " > " + f.value());
+    } else if (filter instanceof LessThan) {
+      LessThan f = (LessThan) filter;
+      return Optional.of(f.attribute() + " < " + f.value());
+    } else if (filter instanceof LessThanOrEqual) {
+      LessThanOrEqual f = (LessThanOrEqual) filter;
+      return Optional.of(f.attribute() + " <= " + f.value());
+    } else if (filter instanceof GreaterThanOrEqual) {
+      GreaterThanOrEqual f = (GreaterThanOrEqual) filter;
+      return Optional.of(f.attribute() + " >= " + f.value());
+    } else if (filter instanceof EqualTo) {
+      EqualTo f = (EqualTo) filter;
+      return Optional.of(f.attribute() + " == " + f.value());
+    } else if (filter instanceof Or) {
+      Or f = (Or) filter;
+      Optional<String> left = compileFilter(f.left());
+      Optional<String> right = compileFilter(f.right());
+      if (left.isEmpty()) return right;
+      if (right.isEmpty()) return left;
+      return Optional.of(String.format("(%s) OR (%s)", left.get(), right.get()));
+    } else if (filter instanceof And) {
+      And f = (And) filter;
+      Optional<String> left = compileFilter(f.left());
+      Optional<String> right = compileFilter(f.right());
+      if (left.isEmpty()) return right;
+      if (right.isEmpty()) return left;
+      return Optional.of(String.format("(%s) AND (%s)",
+          left.get(), right.get()));
+    } else if (filter instanceof IsNull) {
+      IsNull f = (IsNull) filter;
+      return Optional.of(String.format("%s IS NULL", f.attribute()));
+    } else if (filter instanceof  IsNotNull) {
+      IsNotNull f = (IsNotNull) filter;
+      return Optional.of(String.format("%s IS NOT NULL", f.attribute()));
+    } else if (filter instanceof Not) {
+      Not f = (Not) filter;
+      Optional<String> child = compileFilter(f.child());
+      if (child.isEmpty()) return child;
+      return Optional.of(String.format("NOT (%s)", child.get()));
+    }
+
+    return Optional.empty();
+  }
+
+  /**
+   * @param filters filters to see if Lance supported the filter push down
+   * @return the accepted push down filters (row 0) and rejected post scan filters (row 1)
+   */
+  public static Filter[][] processFilters(Filter[] filters) {
+    List<Filter> acceptedFilters = new ArrayList<>();
+    List<Filter> rejectedFilters = new ArrayList<>();
+
+    for (Filter filter : filters) {
+      if (isFilterSupported(filter)) {
+        acceptedFilters.add(filter);
+      } else {
+        rejectedFilters.add(filter);
+      }
+    }
+
+    Filter[] acceptedArray = acceptedFilters.toArray(new Filter[0]);
+    Filter[] rejectedArray = rejectedFilters.toArray(new Filter[0]);
+
+    return new Filter[][]{acceptedArray, rejectedArray};
+  }
+
+  public static boolean isFilterSupported(Filter filter) {
+    if (filter instanceof EqualTo) {
```

Comment:
should we considering using `switch case `?

Reply:
unfortunately, Java currently does not support direct pattern matching with instanceof within a switch statement in a way that allows for case-specific handling as done with instanceof checks